config/file_config.go

```diff
@@ -1,89 +1,108 @@
-	HoneycombAPI              string        `validate:"required,url"`
-	Logger                    string        `validate:"required,oneof= logrus honeycomb"`
-	LoggingLevel              string        `validate:"required"`
-	Collector                 string        `validate:"required,oneof= InMemCollector"`
-	Sampler                   string        `validate:"required,oneof= DeterministicSampler DynamicSampler EMADynamicSampler RulesBasedSampler TotalThroughputSampler"`
-	Metrics                   string        `validate:"required,oneof= prometheus honeycomb"`
-	SendDelay                 time.Duration `validate:"required"`
-	BatchTimeout              time.Duration
-	TraceTimeout              time.Duration `validate:"required"`
-	MaxBatchSize              uint          `validate:"required"`
-	SendTicker                time.Duration `validate:"required"`
-	UpstreamBufferSize        int           `validate:"required"`
-	PeerBufferSize            int           `validate:"required"`
-	DebugServiceAddr          string
-	DryRun                    bool
-	DryRunFieldName           string
+	ListenAddr                string                         `default:"0.0.0.0:8080" cmdenv:"HTTPListenAddr" validate:"required"`
+	PeerListenAddr            string                         `default:"0.0.0.0:8081" cmdenv:"PeerListenAddr" validate:"required"`
+	CompressPeerCommunication bool                           `default:"true"`
+	GRPCListenAddr            string                         `cmdenv:"GRPCListenAddr"`
+	APIKeys                   []string                       `default:"[\"*\"]" validate:"required"`
+	HoneycombAPI              string                         `default:"https://api.honeycomb.io" cmdenv:"HoneycombAPI" validate:"required,url"`
+	Logger                    string                         `default:"logrus" validate:"required,oneof= logrus honeycomb"`
+	LoggingLevel              string                         `default:"info" validate:"required,oneof= debug info warn error"`
+	Collector                 string                         `default:"InMemCollector" validate:"required,oneof= InMemCollector"`
+	Metrics                   string                         `default:"honeycomb" validate:"required,oneof= prometheus honeycomb"`
+	SendDelay                 Duration                       `default:"2s" validate:"required"`
+	BatchTimeout              Duration                       `default:"100ms"`
+	TraceTimeout              Duration                       `default:"60s" validate:"required"`
+	MaxBatchSize              uint                           `default:"500" validate:"required"`
+	SendTicker                Duration                       `default:"100ms" validate:"required"`
+	UpstreamBufferSize        int                            `default:"10_000" validate:"required"`
+	PeerBufferSize            int                            `default:"10_000" validate:"required"`
+	DebugServiceAddr          string                         ``
```

Comment:
I do this because without them, gofmt doesn't align the struct tags across the empty fields. It makes it a lot easier to read the struct tags this way. 

It may be that in the next round, we move them into groups where they can be trailing fields and not break up the struct tags. For now I'd prefer to leave them this way.